cirq-core/cirq/transformers/align.py

```diff
@@ -70,4 +71,6 @@ def align_right(
     Returns:
           Copy of the transformed input circuit.
     """
+    if context is not None and context.deep is True:
+        context = dataclasses.replace(context, deep=False)
```

Comment:
It's because `align_right` calls `align_left` below, which also supports the deep=True flag. If we don't modify the context, then a single call to align_left will left align all nested sub-circuits as well, which we don't want. We want the call to align_left below to only act on the top-level circuit and `align_right` will be called recursively on all sub-circuits by the decorator.